lib/ui/shared/products_grid.dart

```diff
@@ -24,55 +42,76 @@ class _ProductsGridState extends State<ProductsGrid> {
               borderRadius: BorderRadius.circular(8),
               color: const Color(0xFFD1EBF4),
               //D1EBF4 //A4D6E7
-              // image: DecorationImage(
-              //   image: AssetImage(widget.image),
-              //   fit: BoxFit.cover,
-              // ),
+              image: DecorationImage(
+                image: AssetImage(
+                  widget.image,
+                ),
+                fit: BoxFit.contain,
+              ),
+            ),
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.start,
+              crossAxisAlignment: CrossAxisAlignment.end,
+              children: [
+                IconButton(
+                  onPressed: () {
+                    saveItem();
+                  },
+                  icon: Icon(
+                    Icons.favorite,
+                    color: isClicked ? const Color(0XFF007FAC) : Colors.white,
+                  ),
+                ),
+              ],
             ),
           ),
           const SizedBox(height: 8.0),
           Column(
             crossAxisAlignment: CrossAxisAlignment.start,
             children: [
               Row(
-                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                mainAxisAlignment: MainAxisAlignment.spaceBetween,
                 children: [
                   Text(
                     widget.productName,
-                    // maxLines: 1,
-                    // overflow: TextOverflow.clip,
                     style: const TextStyle(
-                      fontSize: 14,
+                      fontSize: 12,
                     ),
                   ),
-                  const Icon(
-                    Icons.star_rate_rounded,
-                    size: 15.0,
-                    color: Colors.amberAccent,
-                  ),
-                  const Icon(
-                    Icons.star_rate_rounded,
-                    size: 15.0,
-                    color: Colors.amberAccent,
-                  ),
-                  const Icon(
-                    Icons.star_rate_rounded,
-                    size: 15.0,
-                    color: Colors.amberAccent,
-                  ),
-                  const Icon(
-                    Icons.star_rate_rounded,
-                    size: 15.0,
-                    color: Colors.amberAccent,
-                  ),
-                  const Icon(
-                    Icons.star_rate_rounded,
-                    size: 15.0,
-                    color: Colors.amberAccent,
-                  ),
+                  const Row(
```

Comment:
Create a reusable widget for this row of stars. Make it such that the size is customizable but default to 12